software_training_assignment/src/goto_action.h

```diff
@@ -0,0 +1,97 @@
+#include <software_training_assignment/GotoAction.h>
+#include <actionlib/server/simple_action_server.h>
+#include <turtlesim/Pose.h>
+#include <geometry_msgs/Twist.h>
+
+namespace A
+{
+    class GotoAction
+    {
+    	public:
+    		ros::NodeHandle node;
+    		GotoAction(std::string name) : as_(node, name, boost::bind(&GotoAction::executeCB, this, _1), false)
```

Comment:
instead of having NodeHandle as public you can pass the NodeHandle into the constructor. Nobody should be modifying the NodeHandle so it should be private